### DISCUSSION TITLE
Tooltip quite always not showing in Windows 10 Enterprise

### DISCUSSION THREAD

**Discussion by capisani** [gnikit/tkinter-tooltip#94] [Q&A]
Trying the first basilar example 

```
import tkinter as tk
import tkinter.ttk as ttk
from tktooltip import ToolTip

app = tk.Tk()
b = ttk.Button(app, text="Button")
b.pack()
ToolTip(b, msg="Hover info")
app.mainloop()
```

with Python 3.12.2, tkinter-tooltip 3.0.0 on Windows 10 Enterprise, tooltips are not displayed in most cases. Sometimes, they appear in random attempts.

**Comment by gnikit** [comment]
I can't reproduce on my local dev environments and with multiple versions of python 3.10-3.12. Have you tried narrowing this down by using other versions of Python or OSs?
I will be converting this to a Q&A discussion.

**Reply by capisani** [reply]
Dear gnikit, 
I am sure that by considering python 3.12 and Windows 10 Enterprise italian version, the problem holds.
Please consider also the discussion on Stack Overflow here
https://stackoverflow.com/questions/78207695/tkinter-tooltip-quite-always-not-showing-in-windows-10-enterprise
where another user found a possible explanation and workaround without using the tooltip library.
Thank you again,
Luca

**Comment by brianzhouzc** [comment]
I'm experiencing the same issue.
Python 3.11.9, tkinter_tooltip-3.1.0. Window 10 Business 22H2 19045.4894

Tooltip does not show up at all using the provided examples.

**Reply by capisani** [reply]
Dear brianzhouzc, 
please see this thread where the problem was solved.

https://stackoverflow.com/questions/78207695/tkinter-tooltip-quite-always-not-showing-in-windows-10-enterprise/78209845?noredirect=1#comment137938275_78209845

Cheers!
Luca

**Reply by brianzhouzc** [reply]
Thanks for the reply. I used the solution provided in the stackoverflow thread for now. Hopefully @gnikit can look into this problem and implement some fixes for the library.

**Reply by toaster-code** [reply]
I think I have the same problem. I managed to find that It does not work properly when delay is set to zero (or not set). I think tkinter need to have some sort of delay to allow triggering the event.
If user set delay=0.1 it works all the time (I am using windows 10).

Here is my code:

```Python
import tkinter as tk
import time
from tktooltip import ToolTip


if __name__ == "__main__":
    root = tk.Tk()
    root.title("Tooltip Example")

    b = tk.Button(root, text="BUTTON A")
    # Create and set up the tooltip
    tooltip = ToolTip(b, msg=time.asctime, delay=0.5)
    b.pack(pady=20)

    b = tk.Button(root, text="BUTTON B")
    # Create and set up the tooltip
    tooltip = ToolTip(b, msg=time.asctime, delay=0.1)
    b.pack(pady=20)

    root.mainloop()
```

**Reply by brianzhouzc** [reply]
Having a delay of 0.1 fixes the issue. Tooltip consistently shows up.

**Comment by brianzhouzc** [comment]
Fixed with PR #107
